four Key Concepts topics (#793)
MicrosoftDocs/quantum-docs-pr: articles/overview/understanding-quantum-computing.md

These quantum particles behave according to the laws of quantum physics, leveraging concepts such as probabilistic computation, superposition, and entanglement, which provide the basis for quantum algorithms that harness the power of quantum computing to solve complex problems. This article describes some of the essential concepts of quantum physics on which quantum computing is based.

## A bird’s-eye view of quantum physics
Quantum physics, also called quantum mechanics or quantum theory, is a branch of physics that deals with particles at the atomic and subatomic level. At the quantum level, however, many of the laws of physics you take for granted don’t apply. Superposition, quantum measurement, and entanglement are three phenomena that are central to quantum computing.  

### Superposition vs. binary computing

> Anyone with a hectic work or school schedule knows that you can’t be two places at once – you have to be either *here* or *there*. However, if you’re a quantum particle then you can be *here* AND *there* AND *somewhere else* due to a phenomenon known as **superposition** (also known as **coherence**).

Reviewer: This description of superposition and coherence is not correct; the "two places at once" analogy is unfortunately misleading, especially for newcomers.

Author: updated the scenario, it should be more accurate.

> Think of flipping a coin in the air and catching it. While it’s in the air turning over and over, it’s neither heads nor tails, but can be seen as a linear combination of all of its possible states at once – heads, tails, 75% heads, 12.5% tails, and so on.

Reviewer: Coherence is something more than classical probability, such that you don't need quantum mechanics to make sense of the spinning coin example.

> A quantum particle such as an electron has its own “heads or tails” property called **spin**, which can be either up or down (or positive or negative for a photon), or to make it more relatable to classical binary computing, let’s just say 1 or 0. When a quantum particle is in a superposition state, it’s a linear combination of all of its possible states between 1 and 0, but you don’t know which one it will be until you actually look at it, which brings up our next phenomenon, **quantum measurement**.

Reviewer: Superposition need not be between all computational basis states; for instance, a two-qubit register can be in the state (|00⟩ + |11⟩) / √2, which has no support on the |01⟩ or |10⟩ basis states.

### Quantum measurement

Let’s say you want to track the height of a five-year old child every year by measuring them and placing a mark on the wall. You would expect their height to increase every year, and the mark on the wall to gradually get higher.

> But what if you measured that five-year old and they just stopped growing? That’s sort of what happens when you measure a quantum particle. Just the act of observing or measuring a quantum particle **collapses** the superposition state (also known as **decoherence**) and the particle takes on a classical binary state of either 1 or 0.

Reviewer: I'm not sure I follow this analogy, sorry?

Author: added new scenario which should make more sense.

This is helpful to us, because in computing you can do lots of things with 1’s and 0’s. However, once a quantum particle has been measured and collapsed, it stays in that state forever and will always be a 1 or 0, and cannot be used for more calculations. As you’ll see later, though, in quantum computing there are operations that can “reset” a particle back to a superposition state so it can be useful again.

### Entanglement

> Possibly the most interesting phenomenon of quantum physics is the ability of two or more quantum particles to become **entangled** with each other. When particles become entangled, they form a single system such that the quantum state of any one particle cannot be described independently of the quantum state of the other particles. This means that whatever operation or process you apply to one particle immediately takes effect on the other particles as well.

Reviewer: > This means that whatever operation or process you apply to one particle immediately takes effect on the other particles as well.

This could be easily misunderstood as claiming that faster-than-light causality is possible, but that's not the case. I'd suggest being a bit more clear about what's meant by "effect" in this case.

> In addition, particles can maintain this connection even when separated over incredibly large distances, even light-years. The effects of quantum measurement still apply to entangled particles, such that when one particle is measured and collapses, the other particle collapses as well. However, the second one always collapses in the *opposite* state as the first one – this particular property is very helpful in quantum computing. These unique properties of entangled particles can be used to transmit quantum information from one place to another by a process called **teleportation**, and may play a critical role in future cryptography applications.

Reviewer: > The effects of quantum measurement still apply to entangled particles, such that when one particle is measured and collapses, the other particle collapses as well. However, the second one always collapses in the *opposite* state as the first one – this particular property is very helpful in quantum computing.

This is not true in general; for example, if two qubits are prepared in the (|00⟩ + |11⟩) / √2 state, then both measurements will always agree with each other. 

> These unique properties of entangled particles can be used to transmit quantum information from one place to another by a process called **teleportation**, and may play a critical role in future cryptography applications.

This seems to conflate QKD with quantum teleportation. While some QKD protocols (e.g.: Ekert94) use teleportation, not all do. The most common one, BB84, doesn't even use multiple qubits.

### Qubits and probability

Classical computers store and process information in bits, which can have a state of either 1 or 0, but never both. The equivalent in quantum computing is the **qubit**, which represents the state of a quantum particle. Because of superposition, qubits can either be 1 or 0, or both, or anything in between. Depending on its configuration, a qubit has a certain *probability* of collapsing to 1 or 0, and this probabilistic state is where the power of quantum computing excels.

In classical computing, uncertainty is not a good thing. In quantum computing, it’s built in.

> For example, with two bits in a classical computer, each bit can store 1 or 0, so together you can store four possible values – **00**, **01**, **10**, and **11** – but only one of those at a time. With two qubits in superposition, however, each qubit can store 1 or 0 or *both*, so you can represent the same four values simultaneously. Furthermore, when multiple entangled qubits are in superposition, they can process multiple options simultaneously. And with each qubit you add to the system, that power increases exponentially.

Reviewer: It is not true in general that the power of a quantum computer increases exponentially with the number of qubits — rather, the difficulty of simulating a quantum computer using a classical computer increases exponentially. Comparing the power of each is significantly more subtle.

---
title: Understanding quantum computing
description: What are quantum computers and how do they use the principles of quantum physics?
author: bradben
ms.author:  bradben
ms.date: 04/22/2020
ms.topic: overview
uid: microsoft.quantum.overview.understanding
---

# Understanding quantum computing

> Quantum computing uses the principles of quantum physics to process information. Because of this, quantum computing requires a different approach than classical computing.  One example of this is the processor used in quantum computers.  Where classical computers use familiar silicon-based chips, quantum computers use quantum particles such atoms, ions, photons, or electrons.  

Reviewer: I'd again suggest here that not all qubits are single particles, such as topological, quantum dot, superconducting, and other modalities of qubits.

## A bird’s-eye view of quantum physics
Quantum physics, also called quantum mechanics or quantum theory, is a branch of physics that deals with particles at the atomic and subatomic levels. At the quantum level, however, many of the laws of physics you take for granted don’t apply. Superposition, quantum measurement, and entanglement are three phenomena that are central to quantum computing.  

### Superposition vs. binary computing

Imagine that you are exercising in your living room. You turn all the way to your left and then all the way to your right. Now turn to your left and your right at the same time. You can’t do it (not without splitting yourself in two, at least).  Obviously, you can’t be in both of those states at once – you can’t be facing left and facing right at the same time.

However, if you were a quantum particle, then you could be *facing left* AND *facing right* AND *anywhere in-between* at the same time due to a phenomenon known as **superposition** (also known as **coherence**).

> A quantum particle such as an electron has its own “facing left or facing right” property called **spin**, which can be either up or down (or positive or negative for a photon), or to make it more relatable to classical binary computing, let’s just say 1 or 0. When a quantum particle is in a superposition state, it’s a linear combination of an infinite number of states between 1 and 0, but you don’t know which one it will be until you actually look at it, which brings up our next phenomenon, **quantum measurement**.

Reviewer: > (or positive or negative for a photon)

Generally 𝑍-axis of the Poincaré sphere (the photonic equivalent of the Bloch sphere) is labeled by |↕⟩ and |↔⟩ rather than "positive" and "negative" or "up" and "down."

Author: I'm trying to keep it very basic here - is the up/down reference too misleading, or can it just be used to make the binary point?

### Quantum measurement

Now, let’s say your friend comes over and wants to take a picture of you exercising. Most likely, they’ll get a blurry image of you turning somewhere between all the way left and all the way right.

But if you’re a quantum particle, an interesting thing happens. No matter where you are when they take the picture, it will always show you either all the way left or all the way right – nothing in-between.

This is because the act of observing or measuring a quantum particle **collapses** the superposition state (also known as **decoherence**) and the particle takes on a classical binary state of either 1 or 0.

> This binary state is helpful to us, because in computing you can do lots of things with 1’s and 0’s. However, once a quantum particle has been measured and collapsed, it stays in that state forever (just like your picture) and will always be a 1 or 0, and cannot be used for more calculations. As you’ll see later, though, in quantum computing there are operations that can “reset” a particle back to a superposition state so it can be useful again.

Reviewer: > and cannot be used for more calculations

The reset process alluded to in the next sentence is precisely performing a measurement; unless a system uses quantum demolition measurement (most common in photonic cases), the qubit is generally left in a known state after measurement that allows us to continue our computation from that known state.

### Entanglement

Possibly the most interesting phenomenon of quantum physics is the ability of two or more quantum particles to become **entangled** with each other. When particles become entangled, they form a single system such that the quantum state of any one particle cannot be described independently of the quantum state of the other particles. This means that whatever operation or process you apply to one particle correlates to the other particles as well, almost instantaneously.

> In addition to this interdependency, particles can maintain this connection even when separated over incredibly large distances, even light-years. The effects of quantum measurement also apply to entangled particles, such that when one particle is measured and collapses, the other particle collapses as well. Because there is a correlation between the entangled qubits, measuring the state of one qubit provides information about the state of the other qubit – this particular property is very helpful in quantum computing. These unique properties of entangled particles can be used to transmit quantum information from one place to another by a process called **teleportation**, and can play a role in some cryptography processes.

Reviewer: As noted in my previous review, teleportation is not used in most QKD protocols; while Ekert91 is teleportation based, BB84 is prepare-and-measure based.

### Qubits and probability

Classical computers store and process information in bits, which can have a state of either 1 or 0, but never both. The equivalent in quantum computing is the **qubit**, which represents the state of a quantum particle. Because of superposition, qubits can either be 1 or 0, or both, or anything in between. Depending on its configuration, a qubit has a certain *probability* of collapsing to 1 or 0, and this probabilistic state is where the power of quantum computing excels.

> In classical computing, uncertainty is not a good thing. In quantum computing, uncertainty is built-in.

Reviewer: Monte Carlo methods can be very useful classically as well. It may be helpful to note that quantum computing isn't just classical computing with randomness.

> For example, with two bits in a classical computer, each bit can store 1 or 0, so together you can store four possible values – **00**, **01**, **10**, and **11** – but only one of those at a time. With two qubits in superposition, however, each qubit can store 1 or 0 or *both*, so you can represent the same four values simultaneously. With three qubits, you can store eight values, with four qubits, you can store 16 values, and so on. Furthermore, when multiple entangled qubits are in superposition, they can process multiple options simultaneously.

Reviewer: Superposition doesn't mean "both 0 and 1," but a linear combination of 0 and 1 ("in-between") as noted earlier in this article.

## Summary

> These concepts just scratch the surface of quantum physics, but are fundamentally important concepts to know for quantum computing.

Reviewer: I'd say most of this isn't physics at all, but rather the basics of quantum mechanics.

---
title: Understanding quantum computing
description: What are quantum computers and how do they use the principles of quantum mechanics?
author: bradben
ms.author:  bradben
ms.date: 5/5/2020
ms.topic: overview
uid: microsoft.quantum.overview.understanding
---

# Understanding quantum computing

> Quantum computing uses the principles of quantum mechanics to process information. Because of this, quantum computing requires a different approach than classical computing.  One example of this difference is the processor used in quantum computers.  Where classical computers use familiar silicon-based chips, quantum computers use quantum material such as atoms, ions, photons, or electrons.  

Reviewer 2: ```suggestion
Quantum computing uses the principles of quantum mechanics to process information. Because of this, quantum computing requires a different approach than classical computing.  One example of this difference is the processor used in quantum computers.  Where classical computers use familiar silicon-based chips, quantum computers use quantum materials such as atoms, ions, photons, or electrons.  
```

The quantum material behaves according to the laws of quantum mechanics, leveraging concepts such as probabilistic computation, superposition, and entanglement. These concepts provide the basis for quantum algorithms that harness the power of quantum computing to solve complex problems. This article describes some of the essential concepts of quantum mechanics on which quantum computing is based.

## A bird’s-eye view of quantum mechanics

> Quantum mechanics, also called quantum theory, is a branch of physics that deals with particles at the atomic and subatomic levels. At the quantum level, however, many of the laws of mechanics you take for granted don’t apply. Superposition, quantum measurement, and entanglement are three phenomena that are central to quantum computing.  

Reviewer 2: I'd argue that interference is a more fundamental phenomena than measurement, as quantum interference determines the result of a measurement. "Collapse" as a result of measurement is an easy escape from explaining what's really happening fundamentally. I would suggest a rewrite of the superposition, measurement and entanglement sections, perhaps if not this time round, definitely soon.

Author: Absolutely, let's revisit this after \\build

### Superposition vs. binary computing

Imagine that you are exercising in your living room. You turn all the way to your left and then all the way to your right. Now turn to your left and your right at the same time. You can’t do it (not without splitting yourself in two, at least).  Obviously, you can’t be in both of those states at once – you can’t be facing left and facing right at the same time.

However, if you are a quantum particle, then you can be *facing left* AND *facing right* AND *anywhere in-between* at the same time due to a phenomenon known as **superposition** (also known as **coherence**).

> A quantum particle such as an electron has its own “facing left or facing right” property called **spin**, referred to as either up or down, or to make it more relatable to classical binary computing, let’s just say 1 or 0. When a quantum particle is in a superposition state, it’s a linear combination of an infinite number of states between 1 and 0, but you don’t know which one it will be until you actually look at it, which brings up our next phenomenon, **quantum measurement**.

Reviewer 2: If we only relate to spin, people will think that 1 or 0 always corresponds to spin up or down. Can we add a few more examples of different kinds of states that can be used as a qubit? Like photons' polarization, a ground state and an excited state of an electron, the topological state of a nanowire, etc... Or changing the phrasing to 
```suggestion
A quantum particle such as an electron has its own “facing left or facing right” properties, for example **spin**, referred to as either up or down, or to make it more relatable to classical binary computing, let’s just say 1 or 0. When a quantum particle is in a superposition state, it’s a linear combination of an infinite number of states between 1 and 0, but you don’t know which one it will be until you actually look at it, which brings up our next phenomenon, **quantum measurement**.
```

### Quantum measurement

Now, let’s say your friend comes over and wants to take a picture of you exercising. Most likely, they’ll get a blurry image of you turning somewhere between all the way left and all the way right.

But if you’re a quantum particle, an interesting thing happens. No matter where you are when they take the picture, it will always show you either all the way left or all the way right – nothing in-between.

> This is because the act of observing or measuring a quantum particle **collapses** the superposition state (also known as **decoherence**) and the particle takes on a classical binary state of either 1 or 0.

Reviewer 2: Interference is a more accepted interpretation than "collapse". Say the quantum particle is in a state sqrt(0.4)|0>+sqrt(0.6)|1>, i.e. you have 40% time facing left and 60% time facing right. Your friend takes 100 pictures. 40 times you are facing left and 60 times you are facing right.

> This is because the act of observing or measuring a quantum particle **collapses** the superposition state (also known as **decoherence**) and the particle takes on a classical binary state of either 1 or 0.

Reviewer 2: ```suggestion
Say, the quantum particle is in a state sqrt(0.4)|0>+sqrt(0.6)|1>, i.e. you have 40% time facing left and 60% time facing right. Your friend takes 100 pictures. 40 times you are facing left and 60 times you are facing right.
```

Author: I added a short introduction to interference in the Probability section that I think helps tie this together.

> This binary state is helpful to us, because in computing you can do lots of things with 1’s and 0’s. However, once a quantum particle has been measured and collapsed, it stays in that state forever (just like your picture) and will always be a 1 or 0. As you’ll see later, though, in quantum computing there are operations that can “reset” a particle back to a superposition state so it can be used for quantum calculations again.

Reviewer 2: I want to avoid misleading readers like every other quantum computing intro. Superposition is not |0> and |1> at the same time. It is a linear combination of |0> and |1>. Interference is a more accepted interpretation than "collapse". Say the quantum particle is in a state sqrt(0.4)|0>+sqrt(0.6)|1>, i.e. you have 40% time facing left and 60% time facing right. Your friend takes 100 pictures. 40 times you are facing left and 60 times you are facing right.

### Entanglement

> Possibly the most interesting phenomenon of quantum mechanics is the ability of two or more quantum particles to become **entangled** with each other. When particles become entangled, they form a single system such that the quantum state of any one particle cannot be described independently of the quantum state of the other particles. This means that whatever operation or process you apply to one particle correlates to the other particles as well, almost instantaneously.

Reviewer 2: ```suggestion
Possibly the most interesting phenomenon of quantum mechanics is the ability of two or more quantum particles to become **entangled** with each other. When particles become entangled, they form a single system such that the quantum state of any one particle cannot be described independently of the quantum state of the other particles. This means that whatever operation or process you apply to one particle correlates to the other particles as well.
```
"Time" is not involved here. Let's avoid this misconception about quantum computing as well.

> In addition to this interdependency, particles can maintain this connection even when separated over incredibly large distances, even light-years. The effects of quantum measurement also apply to entangled particles, such that when one particle is measured and collapses, the other particle collapses as well. Because there is a correlation between the entangled qubits, measuring the state of one qubit provides information about the state of the other qubit – this particular property is very helpful in quantum computing. These unique properties of entangled particles can be used to transmit quantum information from one place to another by a process called **teleportation**.

Reviewer 2: We don't ever use teleportation anywhere in computation algorithms. But we always use entanglement in algorithms. I think the problem here is that it reads as if teleportation is the only useful thing, instead of entanglement.

### Qubits and probability

Classical computers store and process information in bits, which can have a state of either 1 or 0, but never both. The equivalent in quantum computing is the **qubit**, which represents the state of a quantum particle. Because of superposition, qubits can either be 1 or 0 or anything in between. Depending on its configuration, a qubit has a certain *probability* of collapsing to 1 or 0, and this probabilistic state is where the power of quantum computing excels.

In classical computing, uncertainty is usually not a good thing. In quantum computing, it's a built-in asset.

> For example, with two bits in a classical computer, each bit can store 1 or 0, so together you can store four possible values – **00**, **01**, **10**, and **11** – but only one of those at a time. With two qubits in superposition, however, each qubit can store 1 or 0 or *both*, so you can represent the same four values simultaneously. With three qubits, you can store eight values, with four qubits, you can store 16 values, and so on. Furthermore, when multiple entangled qubits are in superposition, they can process multiple options simultaneously.

Reviewer 2: We are not storing information and it is not parallel computing.

```suggestion
For example, with two bits in a classical computer, each bit can represent 1 or 0, so together you can store four possible values – **00**, **01**, **10**, and **11** – but only one of those at a time. With two qubits in superposition, however, each qubit has a probability to be a 1 or 0. So you can represent the same four values. With three qubits, there are eight possible values; with four qubits, there are 16, and so on. The number of possible values grow exponentially with the number of qubits. 
```

## Summary

These concepts just scratch the surface of quantum mechanics, but are fundamentally important concepts to know for quantum computing.

> - **Superposition** - The ability of quantum particles to hold multiple states at once.

Reviewer 2: ```suggestion
- **Superposition** - The ability of quantum particles to be a combination of all possible states.
```

> - **Quantum measurement** - The act of observing a quantum particle in superposition and collapsing it to a binary state.

Reviewer 2: ```suggestion
- **Quantum measurement** - The act of observing a quantum particle in superposition and collapsing it to one of the possible states.
```

> - **Entanglement** - The ability of quantum particles to bond and behave as a single system.

Reviewer 2: ```suggestion
- **Entanglement** - The ability of quantum particles to correlate their measurement results to each other.
```

Reviewer 2: Both entanglement and interference (which determines the result of measurement) are results of superposition.

> - **Qubit** - The basic unit of information in quantum computing. A qubit represents a quantum particle and can store multiple states simultaneously.

Reviewer 2: ```suggestion
- **Qubit** - The basic unit of information in quantum computing. A qubit represents a quantum particle in superposition of all possible states.
```

Reviewer 2: Both entanglement and interference (which determines the result of measurement) are results of superposition.

### Entanglement

Possibly the most interesting phenomenon of quantum mechanics is the ability of two or more quantum particles to become **entangled** with each other. When particles become entangled, they form a single system such that the quantum state of any one particle cannot be described independently of the quantum state of the other particles. This means that whatever operation or process you apply to one particle correlates to the other particles as well.

In addition to this interdependency, particles can maintain this connection even when separated over incredibly large distances, even light-years. The effects of quantum measurement also apply to entangled particles, such that when one particle is measured and collapses, the other particle collapses as well. Because there is a correlation between the entangled qubits, measuring the state of one qubit provides information about the state of the other qubit – this particular property is very helpful in quantum computing.

### Qubits and probability

> Classical computers store and process information in bits, which can have a state of either 1 or 0, but never both. The equivalent in quantum computing is the **qubit**, which represents the state of a quantum particle. Because of superposition, qubits can either be 1 or 0 or anything in between. Depending on its configuration, a qubit has a certain *probability* of collapsing to 1 or 0. The ability to configure a qubit's probability of collapsing one way or the other is called **quantum interference**. 

Reviewer 2: ```suggestion
Classical computers store and process information in bits, which can have a state of either 1 or 0, but never both. The equivalent in quantum computing is the **qubit**, which represents the state of a quantum particle. Because of superposition, qubits can either be 1 or 0 or anything in between. Depending on its configuration, a qubit has a certain *probability* of collapsing to 1 or 0. The ability to configure a qubit's probability of collapsing one way or the other is determined by **quantum interference**. 
```

For example, with two bits in a classical computer, each bit can store 1 or 0, so together you can store four possible values – **00**, **01**, **10**, and **11** – but only one of those at a time. With two qubits in superposition, however, each qubit can be 1 or 0 or *both*, so you can represent the same four values simultaneously. With three qubits, you can represent eight values, with four qubits, you can represent 16 values, and so on.

## Summary

These concepts just scratch the surface of quantum mechanics, but are fundamentally important concepts to know for quantum computing.

- **Superposition** - The ability of quantum particles to be a combination of all possible states.
- **Quantum measurement** - The act of observing a quantum particle in superposition and resulting in one of the possible states.
- **Entanglement** - The ability of quantum particles to correlate their measurement results with each other.
- **Qubit** - The basic unit of information in quantum computing. A qubit represents a quantum particle in superposition of all possible states.

> - **Interference** - Actions applied to a qubit to influence the probability of it collapsing one way or another.

Reviewer 2: ```suggestion
- **Interference** - Intrinsic behavior of a qubit due to superposition to influence the probability of it collapsing one way or another.
```
